Improve support for unioned queries on MySQL.

markstory opened the pull request:
By wrapping unioned queries with () each arm in an union can have an order clause. I've created a specialized compiler for SQLite as it seems to be the only rdbms that does not support () in UNION from what I could see.

Refs #5962

markstory committed:
Improve support for unioned queries on MySQL.

By wrapping unioned queries with () each arm in an union can have an
order clause. I've created a specialized compiler for SQLite as it seems
to be the only rdbms that does not support () in UNION from what I could
see.

Refs #5962

markstory committed:
Fix missing () in unioned queries.

The original SELECT also needs () for unions to work. While having the
parentheses added in very disparate parts of the code is a bit awkward
I really didn't want to restructure the entire QueryCompiler for this
scenario.

lorenzo commented:
I think one of the tests is making sqlserver unhappy

lorenzo commented on tests/TestCase/Database/QueryTest.php:
```diff
@@ -1945,6 +1945,31 @@ public function testUnion()
+            ->union($union)
+            ->execute();
+        $this->assertCount(self::COMMENT_COUNT + self::ARTICLE_COUNT, $result);
+    }
```
missing `closeCursor()` here, I think this is why SQLServer is whining

markstory committed:
Close the cursor after checking the count.

markstory commented:
SQLServer like SQLite does not support `ORDER BY` in the queries used in an `UNION`. If you want to order a UNION in both you have to use `SELECT * FROM ( <union>) AS results ORDER BY results.column`. I'll exclude SQLserver from this new behavior as well.

markstory committed:
Make union not supporting order by a flag.

There are 2 dialects that do not support this behavior and I felt bad
creating a trait or duplicating code for this.

markstory commented:
AppVeyor and Travis passing now.
